ibm/resource_ibm_is_snapshot.go

```diff
@@ -0,0 +1,483 @@
+			},
+
+			isSnapshotSourceImage: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "If present, the image id from which the data on this volume was most directly provisioned.",
+			},
+
+			isSnapshotOperatingSystem: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The globally unique name for the operating system included in this image",
+			},
+
+			isSnapshotBootable: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates if a boot volume attachment can be created with a volume created from this snapshot",
+			},
+
+			isSnapshotLCState: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Snapshot lifecycle state",
+			},
+			isSnapshotCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+			isSnapshotEncryption: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Encryption type of the snapshot",
+			},
+			isSnapshotEncryptionKey: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A reference to the root key used to wrap the data encryption key for the source volume.",
+			},
+
+			isSnapshotHref: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "URL for the snapshot",
+			},
+
+			isSnapshotMinCapacity: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Minimum capacity of the snapshot",
+			},
+			isSnapshotResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type of the snapshot",
+			},
+
+			isSnapshotSize: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The size of the snapshot",
+			},
+		},
+	}
+}
+
+func resourceIBMISSnapshotValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 isSnapshotName,
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Required:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$`,
+			MinValueLength:             1,
+			MaxValueLength:             63})
+	ibmISSnapshotResourceValidator := ResourceValidator{ResourceName: "ibm_is_snapshot", Schema: validateSchema}
+	return &ibmISSnapshotResourceValidator
+}
+
+func resourceIBMISSnapshotCreate(d *schema.ResourceData, meta interface{}) error {
+	sv := ""
+	rg := ""
+	name := ""
+	if sourceVolume, ok := d.GetOk(isSnapshotSourceVolume); ok {
+		sv = sourceVolume.(string)
+	}
+	if grp, ok := d.GetOk(isVPCResourceGroup); ok {
+		rg = grp.(string)
+	}
+	log.Printf("[DEBUG] Snapshot create")
+	if snapshotName, ok := d.GetOk(isSnapshotName); ok {
+		name = snapshotName.(string)
+	}
+	err := snapshotCreate(d, meta, name, sv, rg)
```

Comment:
I don't think  snapshotCreate seperated method is required here...and you can eliminate declaration of 
	sv := ""
	rg := ""
	name := ""
all these variables